docs/data-sources/sidecar_instance.md

```diff
@@ -0,0 +1,64 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "cyral_sidecar_instance Data Source - terraform-provider-cyral"
+subcategory: ""
+description: |-
+  Retrieve sidecar instances.
+---
+
+# cyral_sidecar_instance (Data Source)
+
+Retrieve sidecar instances.
+
+<!-- schema generated by tfplugindocs -->
+
+## Schema
+
+### Required
+
+- `sidecar_id` (String) Sidecar identifier.
+
+### Read-Only
+
+- `id` (String) Data source identifier.
+- `instance_list` (List of Object) List of existing sidecar instances. (see [below for nested schema](#nestedatt--instance_list))
+
+<a id="nestedatt--instance_list"></a>
+
+### Nested Schema for `instance_list`
+
+Read-Only:
+
+- `id` (String)
+- `metadata` (Set of Object) (see [below for nested schema](#nestedobjatt--instance_list--metadata))
+- `monitoring` (Set of Object) (see [below for nested schema](#nestedobjatt--instance_list--monitoring))
+
+<a id="nestedobjatt--instance_list--metadata"></a>
+
+### Nested Schema for `instance_list.metadata`
+
+Read-Only:
+
+- `capabilities` (Set of Object) (see [below for nested schema](#nestedobjatt--instance_list--metadata--capabilities))
+- `dynamic_version` (Boolean)
+- `last_registration` (String)
+- `recycling` (Boolean)
+- `start_timestamp` (String)
+- `version` (String)
+
+<a id="nestedobjatt--instance_list--metadata--capabilities"></a>
+
+### Nested Schema for `instance_list.metadata.capabilities`
+
+Read-Only:
+
+- `recyclable` (Boolean)
+
+<a id="nestedobjatt--instance_list--monitoring"></a>
+
+### Nested Schema for `instance_list.monitoring`
+
+Read-Only:
+
+- `services` (Map of Set of Object)
```

Comment:
The `tfplugindocs` seems to be very limited in generating descriptions for nested fields, especially for TypeMap, since not even the map structure is being generated. I still need to come up with a solution for that so that the docs are correctly generated, but other than that, the PR should be good for review.

Reply:
Had to generate the docs manually for this data source. [Update docs](https://github.com/cyralinc/terraform-provider-cyral/pull/457/commits/998fa05d9ec62ca88b88e1e00b582b1a170fd535).